action.yml

```diff
@@ -0,0 +1,106 @@
+name: Terraform Cloud Workspace
+description: Manages Terraform Cloud workspaces
+inputs:
+  terraform-version:
+    description: Terraform version
+    default: "1.0.3"
+  terraform-token:
+    description: Terraform Cloud token
+    required: true
+  terraform-host:
+    description: Terraform Cloud host
+    default: app.terraform.io
+  terraform-organization:
+    description: Terraform Cloud organization
+    required: true
+  name:
+    description: Name of the workspace
+    default: "${{ github.event.repository.name }}"
+  runner-terraform-version:
+    description: Terraform version used to create the workspace
+    default: "1.0.3"
+  environments:
+    description: Environments to create multiple workspaces
+    default: ""
+  aws-role:
```

Comment:
This couples very tightly to AWS. The user should be able to customize the backend config but should provide the full set of keys/values. As JSON or whatever else is convenient.

Reply:
yep agreed, mentioned this in the description but haven't put it into action yet. typings for gh action inputs are rudimentary, but no reason we couldnt try and parse the input as a json field